### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by satazor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,220 @@
...
+$ npm install @moxy/next-layout
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+Next.js projects usually have the need to have one or more layouts. Layouts are the "shell" of your app and usually contain navigation elements, such as an header and a footer. In the ideal scenario, each page would be able to say which layout they want to use, including tweaking its properties dynamically, such as `variant="light"`. However, we also want to keep the layout persistent in the React tree, to avoid having to remount it every time a user navigate between pages.
+
+Historically, projects overlook the need of multiple layouts or the ability to change layout props between pages. They start off with a simple layout and only later they handle this need, often with poor and non-scalable solutions.
```
Comment (acostalima) [inline README.md:31]:
```suggestion
Historically, projects have been overlooking the need of multiple layouts or the ability to change layout props between pages. They start off with a simple layout and only later they handle this need, often with poor and non-scalable solutions.
```

File: src/with-layout.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const layoutSymbol = Symbol('layout');
+
+const withLayout = (mapToLayout) => (Component) => {
+    const value = typeof layout === 'function' ? mapToLayout : () => mapToLayout;
```
Comment (acostalima) [inline src/with-layout.js]:
```suggestion
    const value = typeof mapToLayout === 'function' ? mapToLayout : () => mapToLayout;
```
Right?

**Review by acostalima — COMMENTED** [review]
Nice job! 💪

File: src/with-layout.js (hunk 2 above)
Comment (satazor) [reply src/with-layout.js]: Nice catch! This was a bug. I've fixed it and add more tests.

**Pull request merged**